src/components/submission/list.vue

```diff
@@ -276,6 +276,13 @@ export default {
       })
         .finally(() => { this.refreshing = false; })
         .catch(noop);
+
+      // refresh keys when refreshing submissions (keys initially loaded in form/submissions)
+      if (refresh) {
+        this.keys.request({
+          url: apiPaths.submissionKeys(this.projectId, this.xmlFormId)
```

Comment:
On the draft testing page, we need to fetch the keys for the draft, not the form. `apiPaths.submissionKeys()` takes an optional third argument for whether the form version is a draft. One place where you can see that is that `FormSubmissions` and `FormDraftTesting` send different requests for `keys`.

Actually, since those two components are already set up to send requests for `keys`, one option is for `SubmissionList` to emit an event here (e.g., `fetch-keys`) to indicate to the parent component that it should re-request `keys`. `FormSubmissions` and `FormDraftTesting` would have to listen for the event. The two components already know how to request `keys` in different ways, so it'd just be a matter of setting up the event. `ProjectShow` and `FormShow` use a similar pattern. If you take that approach though, you won't be able to use `loadSubmissionList()` to test this functionality, since it only mounts `SubmissionList`, not `FormSubmissions` or `FormDraftTesting`. You could use `load()` instead though.

Reply:
I tried this other approach, of emitting an event and having the parent component deal with it, and i like it. I'm not that sure how to test it. 

I added one test that emulates the "quirk" of the backend returning `[]` for keys when there are 0 submissions (but otherwise a key should probably be returned).